docker/Dockerfile.nixl

```diff
@@ -0,0 +1,176 @@
+ARG BASE_IMAGE="nvcr.io/nvidia/cuda"
+ARG BASE_IMAGE_TAG="12.5.1-devel-ubuntu24.04"
+
+ARG RUNTIME_IMAGE="nvcr.io/nvidia/cuda"
+ARG RUNTIME_IMAGE_TAG="12.5.1-devel-ubuntu24.04"
+
+FROM ${BASE_IMAGE}:${BASE_IMAGE_TAG} AS nixl_base
+ENV DEBIAN_FRONTEND=noninteractive
+RUN apt-get update -y && apt-get install git -y
+
+WORKDIR /opt
+RUN git clone https://github.com/ai-dynamo/nixl.git
+WORKDIR /opt/nixl
+
+# Add a cache hint that only changes when the nixl commit changes
+ARG NIXL_COMMIT="53b87ebd007aeb14e5ba7ff602d2e92977d66aad"
```

Comment:
Should we use [0.1.1](https://github.com/ai-dynamo/nixl/releases/tag/0.1.1) release instead?